robot-agent/packageMonitor.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs');
+
+const constants = require('./constants');
+const { spawn, exec, execSync } = require('child_process');
+const { EventEmitter } = require('events');
+
+const { getLogger } = require('@transitive-sdk/utils');
+const log = getLogger('packageMonitor.js');
+log.setLevel('info');
+
+class PackageMonitor extends EventEmitter {
```

Comment:
The big thing missing from this class is any info from `runningPackages` (or it's original source, namely clients connecting to the local mqtt broker). That is by far the most reliable source of truth about whether the package is running or not. So I find it confusing to introduce things like `isRunning` in this class that take a different meaning (namely whether we can find the PID).